File: lib_java_test/src/main/java/com/example/lib_java_test/ZhengZeTest.java
package com.example.lib_java_test;

public class ZhengZeTest {
    public static void main(String[] args) {
//        matchStr()
        String firstPart = "/comment/";
        String secondPart = "/detail";
        System.out.println(matchStr("sfasfsafsaf/comment/12131/detailsfsafsafasasf ",
                ".*" + firstPart + "\\w*" + secondPart + ".*"));
    }

    public static boolean matchStr(String str,String reg){
        return str.matches(reg);
    }

}

File: app/src/main/java/com/example/zhangshuai/test/ZhengZeTest.kt
package com.example.zhangshuai.test

object ZhengZeTest {
    @JvmStatic
    fun main(args: Array<String>) {
//        matchStr()
        val firstPart = "/comment/"
        val secondPart = "/detail"
        println(matchStr("sfasfsafsaf/comment/12131/detailsfsafsafasasf ",
                ".*$firstPart\\w*$secondPart.*"))
    }

    fun test(){
        val firstPart = "/comment/"
        val secondPart = "/detail"
        println(matchStr("sfasfsafsaf/comment/12131/detailsfsafsafasasf ",
                ".*$firstPart\\w*$secondPart.*"))
    }

    fun matchStr(str: String, reg: String?): Boolean {
        return str.matches(reg!!.toRegex())
    }
}